server/plugin.go

```diff
@@ -74,37 +86,107 @@ func (p *Plugin) OnActivate() error {
 	}
 	p.BotUserID = botID
 
-	channel, appErr := p.API.GetChannelByName(team.Id, channelSplit, false)
+	p.API.LogInfo("bot user", "id", p.BotUserID)
+
+	// get or create channel and return mattermost channel id
+	err = p.getOrCreateMattermostChannels(teamChannels)
+	if err != nil {
+		return err
+	}
+
+	p.Channels = teamChannels
+
+	p.API.LogInfo("channels resvolved", "tc", teamChannels)
+	if err := p.registerCommands(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Plugin) resolveAndSetTeamIDs(channels []*TeamChannel) error {
+	//mattermostChannels := []TeamChannel{}
+	for _, teamChannel := range channels {
+		p.API.LogInfo("resolve for teamchannel", "tc", teamChannel)
+		appErr := p.resolveAndSetTeamID(teamChannel)
+		if appErr != nil {
+			return appErr
+		}
+		//mattermostChannels = append(mattermostChannels, *teamChannel)
+	}
+	return nil
+}
+
+func (p *Plugin) resolveAndSetTeamID(teamChannel *TeamChannel) error {
+	team, appErr := p.API.GetTeamByName(teamChannel.TeamName)
+	if appErr != nil {
+		return appErr
+	}
+	teamChannel.TeamID = team.Id
+	return nil
+}
```

Comment:
Considering how small this method is and it's only used in one place (`resolveAndSetTeamIDs`, also a small method), I think we can remove this method and have this block inside of `resolveAndSetTeamIDs` instead